test/functoria-test/functoria_test.ml

```diff
@@ -17,8 +17,8 @@ let run ?(keys = []) ?init context device =
   let keys = keys @ Key.Set.elements (Engine.all_keys t) in
   let packages = Key.eval context (Engine.packages t) in
   let info =
-    Functoria.Info.v ~packages ~context ~keys ~build_cmd:[ "build"; "me" ]
-      ~src:`None "foo"
+    Functoria.Info.v ~config_file:(Fpath.v "config.ml") ~packages ~context ~keys
```

Comment:
Do you still need this now that we have a default argument?

Reply:
Yes, what I did in [8128e28](https://github.com/mirage/mirage/pull/1309/commits/8128e28fcb562beaed588ff3660cfbbcabb7d7a7) is about `Functoria.Lib.v`, not `Functoria.Info.v` which does not define optional values.

Reply:
could you define an optional value there too? That way we don't have to break the API.

Reply:
Hmmhmm, I don't have strong opinion about that but it seems clear that any arguments expected by `Functoria.Lib.v` are required (and so non-optional). It seems spontaneous to stick on this design.

Reply:
I'm even tempted to force all the calls to this function to strip out the directory component. I remember why I removed it from the 3.* design (as well as the build-dir option): it was really unclear what that directory component is about. I think we should make this clearer in the API:
- always keep `config_file` as an optional argument with default being "config.ml": we don't expect people to change this at all.
- always check that users of that functions that are providing a custom `config_file` should call it with an empty directory (otherwise the semantics is really unclear)

Ideally add a new test-case for this. I really remember that something was a bit tricky about this (maybe the interaction between `--build-dir` and `--config-file`).

Reply:
I don't see where `build-dir` is defined (as a command-line argument). From what I see, `build_dir = Fpath.parent args.Cli.config_file`. Can you describe which workflow can be problematic?